add rfc for modernizing learn js

chrisdavidmills opened the pull request:
<!-- Please provide the following information to help us review this PR: -->

> What was wrong/why is this fix needed? (quick summary only)

This is an RFC to describe what should be done about modernizing the [JavaScript modules](https://developer.mozilla.org/en-US/docs/Learn/JavaScript) in the learning area.

> MDN URL of main page changed

https://developer.mozilla.org/en-US/docs/Learn/JavaScript

> Issue number (if there is an associated issue)

n/a

> Anything else that could help us review it

n/a

chrisdavidmills committed:
add rfc for modernizing learn js

chrisdavidmills commented:
Added @wbamberg as a reviewer; it'd also be lovely to get some input from @jadjoubran on this ;-)

schalkneethling reviewed (approved):
This all sounds great to me @chrisdavidmills

wbamberg commented on rfcs/modernize-learn-js.md:
```diff
@@ -0,0 +1,67 @@
+
+## Task list
+
+- Agree on an overall strategy for the updates. For example, is there a
```
We have https://developer.mozilla.org/en-US/docs/MDN/Guidelines/Code_guidelines/JavaScript and if possible it would be great to have a common set of guidelines across MDN.

That guide currently advises against using async/await, perhaps we could change that now?

escattone replied:
@wbamberg 
> That guide currently advises against using async/await, perhaps we could change that now?

Definitely! The use of `async/await` is fully/naturally supported!

chrisdavidmills replied:
Good call. I've sneaked in an update to that guideline to this PR ;-)

wbamberg commented on rfcs/modernize-learn-js.md:
```diff
@@ -0,0 +1,67 @@
+  resource? How new is too new? Do we gate it on what is supported in at least
+  two rendering engines, or somesuchthing? How do we deal with
+  backwards-compatibility? Show examples of how to support older browsers, or
+  just teach everyone how to use Babel.js to begin with?
```
It seems like it would be helpful to have a guide (or part of a guide) early on about targeting browsers, where we could say things like:

* when you write JS you always need to consider which browsers/non-browser runtimes you need to support
* in this guide we use [the following JS standards] which will work with [these reasonably modern engines] but in particular will not always work with [those ancient engines]
* if you do need to support [those ancient engines] there are tools like Babel

chrisdavidmills replied:
great idea; I've added it as a suggestion to the RFC

wbamberg reviewed (approved):
Looks great to me @chrisdavidmills , thanks for filing it!

jadjoubran commented:
Very nice idea @chrisdavidmills! I think that even something as basic as changing `var` to `let`/`const` would already help learners a lot, and this seems to be going in the right direction with more planned improvements. Happy to take part of the auditing/fixing process

chrisdavidmills committed:
add suggestions, update async/await guideline

chrisdavidmills commented:
> Very nice idea @chrisdavidmills! I think that even something as basic as changing `var` to `let`/`const` would already help learners a lot, and this seems to be going in the right direction with more planned improvements. Happy to take part of the auditing/fixing process

Thanks for your interest @jadjoubran ! We already updated quite a lot of it to use `const`/`let`, but I've added in a reminder to the RFC anyway, to check if there is anything else to do in that regard.
